### DISCUSSION TITLE
Overlay by "intersect" is dropping entire polygons that should have instersections

### DISCUSSION THREAD

**Discussion by martincec** [geopandas/geopandas#3448] [Q&A]
I am trying to create a crosswalk between California zip codes and an older version of Census tracts by using Overlay by "intersection". The result keeps dropping large areas that should not be dropped though. 

``` 
import pandas as pd
import shapely
import geopandas as gpd
import matplotlib.pyplot as plt

census_tracts = gpd.GeoDataFrame.from_file("tracts.shp")
ZIP = gpd.GeoDataFrame.from_file("California_Zip_Codes.shp")

proj = 3311 # NAD83/HARN Projection EPSG (units in meters)

census_tracts = census_tracts .to_crs(proj)
ZIP = ZIP.to_crs(proj)

crosswalk = gpd.overlay(ZIP, census_tracts , how="intersection")
```

** Note: the entire state should be filled. 
![image](https://github.com/user-attachments/assets/53141183-23a5-4b3f-861c-4c6861d7aaba)

**Comment by theroggy** [comment]
Typically this is due to some polygons being invalid. Could you checked if all polygons are valid with `is_valid` and if not to use `make_valid` to fix them?
